.gitlab-ci.yml

```diff
@@ -0,0 +1,89 @@
+image: hugomrdias/node-alpine:test
+stages:
+  - check
+  - test
+  - cov
+
+variables:
+  DOCKER_DRIVER: overlay2
+
+before_script:
+  - npm i
+
+check:
+  stage: check
+  script:
+    - npm run lint
+    - npm run dep-check
+    # - npm install --no-lockfile @commitlint/config-conventional @commitlint/cli --save-dev
+    # - npx commitlint --extends=@commitlint/config-conventional --from=$CI_COMMIT_BEFORE_SHA --to=$CI_COMMIT_SHA
+    - npm run build
+  artifacts:
+    paths:
+      - dist/
+
+linux-node-10:
+  image: node:10
+  stage: test
+  script:
+    - npx nyc -s npm run test:node -- --bail
+  artifacts:
+    paths:
+      - .nyc_output/
+
+macos-node-10:
+  stage: test
+  script:
+    - npx nyc -s npm run test:node -- --bail
+  tags:
+    - mac
+  artifacts:
+    paths:
+      - .nyc_output/
+
+win-node-10:
+  stage: test
+  script:
+    - npx nyc -s npm run test:node -- --bail
+  tags:
+    - win
+  artifacts:
+    paths:
+      - .nyc_output/
+
+test-browser-chrome:
+  image: hugomrdias/node-chrome:test
+  stage: test
+  script:
+    - AEGIR_BROWSERS=ChromeDocker npm run test:browser
+    - AEGIR_BROWSERS=ChromeDocker npm run test:webworker
+  artifacts:
+    paths:
+      - coverage/
+
+test-browser-firefox:
+  allow_failure: true
```

Comment:
Does this meant that the overall build will pass even if this step fails? What's the reason for adding this currently?